lms/djangoapps/certificates/apis/v0/views.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.core.lib.api import permissions
+from lms.djangoapps.certificates.api import get_certificate_for_user
+
+log = logging.getLogger(__name__)
+
+
+class CertificatesDetailView(GenericAPIView):
+    """
+        **Use Case**
+
+            * Get the details of a certificate for a specific user in a course.
+
+        **Example Request**
+
+            GET /api/certificates/v0/certificates/{username}/{course_id}
+
+        **GET Parameters**
+
+            A GET request must include the following parameters.
+
+            * username: A string representation of an user's username.
+            * course_id: A string representation of a Course ID.
+
+        **GET Response Values**
+
+            If the request for information about the Certificate is successful, an HTTP 200 "OK" response
+            is returned.
+
+            The HTTP 200 response has the following values.
+
+            * username: A string representation of an user's username passed in the request.
+
+            * course_id: A string representation of a Course ID.
+
+            * certificate_type: A string representation of the certificate type.
+                Can be honor|verified|professional
+
+            * status: A string representation of the certificate status.
+
+            * download_url: A string representation of the certificate pdf url.
+
+            * grade: A string representation of a float for the user's course grade.
+
+        **Example GET Response**
+
+            {
+                "username": "bob",
+                "course_id": "edX/DemoX/Demo_Course",
+                "certificate_type": "verified",
+                "status": "downloadable",
+                "download_url": "http://www.example.com/cert.pdf",
+                "grade": "0.98"
+            }
+    """
+
+    authentication_classes = (OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated, permissions.IsUserInUrlOrStaff)
+
+    def get(self, request, username, course_id):
+        """
+        Gets a certificate information.
+
+        Args:
+            request (Request): Django request object.
+            username (string): URI element specifying the user's username.
+            course_id (string): URI element specifying the course location.
+
+        Return:
+            A JSON serialized representation of the certificate.
+        """
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            log.info('Course ID string "%s" is not valid', course_id)
+            return Response(
+                status=404,
+                data={
+                    'error_code': 'Course ID string "{}" is not valid'.format(course_id)
```

Comment:
This 'error_code' value looks more like it belongs in the 'developer_message' category, according to the [guidelines](https://openedx.atlassian.net/wiki/display/AC/edX+REST+API+Conventions#edXRESTAPIConventions-5.Errors).  The error_code, if used, should be a unique identifier for the type of error, rather than a detailed description.  On the other hand, if it's actually meant to be a user-facing message it will have to be i18n-ed.

Same comment applies to the next error response below.